File: edit_distance/security_dp.py
import math

import matplotlib.pyplot as plt
import numpy as np

M = 64
points = np.zeros(M + 1)
lowers = np.zeros(M + 1)
for m in range(len(lowers)):
    lowers[m] = pow(2, m)
    points[m] = math.factorial(m)

plt.close()
plt.figure()
plt.grid(linestyle="--")
plt.xlabel("$M$", fontsize=12, fontweight='bold')
plt.ylabel("The number of guesses", fontsize=12, fontweight='bold')
plt.yscale('log')

plt.plot(range(len(points)), points, linestyle='--', color="k", linewidth=2, label="$M_{guess}^{BM}$")
plt.plot(range(len(lowers)), lowers, color="r", linewidth=2, label="$2^M$")
plt.ylim(1.66223336e-11, 1.268869e+97)

plt.legend(loc='upper left')
leg_text = plt.gca().get_legend().get_texts()
plt.setp(leg_text, fontsize=10, fontweight='bold')
plt.savefig('./evaluations/nAllPlot_dp.pdf', format='pdf')
plt.show()

File: sorting_index/plot/no additive nosie/security_dp.m
clear; close all; clc
textSize = 14;
linewid = 1;

points = [];
lowers = [];
for i=1:65
    lowers(i) = 2^(i-1);
    points(i) = factorial(i-1);
end
figure(1)
plot([1:65], points, 'k--', 'LineWidth',2)
hold on
plot([1:65], lowers, 'r-', 'LineWidth',2)
hold on

ax = gca;
ax.YAxis.FontSize = 13;
ax.XAxis.FontSize = 13;
ax.YAxis.FontWeight = 'bold';
ax.XAxis.FontWeight = 'bold';
ax.YScale='log';
ax.YTick=[0, 1e+20, 1e+40, 1e+60, 1e+80, 1e+100, 1e+120];

ylim([0, 1e+120]);
legend('M_{guess}^{BM}','2^M', 'FontSize',textSize, 'FontWeight','bold', 'Location','northwest')
xlabel('M', 'FontSize', 15, 'FontWeight','bold')
ylabel('The number of guesses', 'FontSize',15, 'FontWeight','bold')
grid on

print('nAllPlot_dp','-depsc')